PhilanthropyDataCommons/service
# Implement Data Validation

This PR introduces basic data validation for uploaded proposal fields. It modifies the base_fields and proposal_field_values tables in the database and related types in the service layer to accommodate 
1. A set of predefined acceptable types for a field
2. A boolean field "is_valid" to indicate if the proposal field's value's type matches the related base field

Validation is performed by a helper function, fieldValueIsValid, which uses regexes, though a more developed validation 
system will likely be necessary as we account for more complicated types. 

Testing:
1. Start pdc service
2. Run migrations. 
  a. NOTE: All base field types will be set to string by default, which are valid by default in the validation system. 
                  In order to test individual types(such as number, phone number, email, etc), new base fields with the 
                  corresponding type will need to be created and inserted into the database.
3. Start pdc frontend, navigate to the add data page
4. Upload a proposal with chosen fields valid or invalid
5. Verify that the expected validity is reflected in the "is_valid" column in the proposal_field_values table in the database


Resolves #760
